### ISSUE TITLE
DB BAP service status API regularly fails with 503 Service unavailable

### ISSUE THREAD

**Issue by abirke** [felix-zenk/onboardapis#2]
Hi there,

it seems to me the DB BAP service status API quite regularly fails with HTTP 503. With `ICEPortal().init()` calling the BAP API this exception leaves all other APIs unusable as well. Have you seen this issue, too? I will attach the exact exception below.

Thanks and best regards

```
WARNING:root:Request to https://iceportal.de/bap/api/bap-service-status returned status code 503
Traceback (most recent call last):
  File "onboard\.venv\Lib\site-packages\requests\models.py", line 971, in json
    return complexjson.loads(self.text, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "AppData\Local\Programs\Python\Python311\Lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "AppData\Local\Programs\Python\Python311\Lib\json\decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "AppData\Local\Programs\Python\Python311\Lib\json\decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "onboard\.venv\Lib\site-packages\onboardapis\utils\data.py", line 459, in _get
    return super(JSONDataConnector, self)._get(endpoint, *args, base_url=base_url, **kwargs).json()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "onboard\.venv\Lib\site-packages\requests\models.py", line 975, in json
    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "onboard\.venv\Lib\site-packages\onboardapis\trains\__init__.py", line 251, in init
    self._static_data.refresh()
  File "onboard\.venv\Lib\site-packages\onboardapis\trains\germany\db.py", line 31, in refresh
    self._get("/bap/api/bap-service-status")
  File "onboard\.venv\Lib\site-packages\onboardapis\utils\data.py", line 462, in _get
    raise DataInvalidError() from e
onboardapis.exceptions.DataInvalidError: The API returned an invalid response

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "onboard\status.py", line 5, in <module>
    train.init()  # Explicit call to init method to initialize API connection
    ^^^^^^^^^^^^
  File "onboard\.venv\Lib\site-packages\onboardapis\trains\__init__.py", line 254, in init
    raise InitialConnectionError() from e
onboardapis.exceptions.InitialConnectionError: Unable to connect to the API, are you connected to the on-board WI-FI?
```

**Comment by felix-zenk** [comment]
Hi,

I've never experienced a 503 with the BAP API. It is however quite new, so I imagine DB is changing things around more or less frequently. I patched the 503 and released a new version. Be warned, that this patch makes any API GETs not null safe anymore, as it only issues a logger warning instead of raising the error.
I believe the patch that I decided on to be sufficient, as I plan to release a new major version soon.
